From f5fec29b9b98f49cdf1834e1a568aef21ab96047 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 30 Sep 2022 11:57:02 +0300
Subject: [PATCH] [SW-96202] gaudi3: errata wa for mem cpy rd os

As a workaround for hw errata GAUDI3_0306, memcpy rd_os register
is set to 0.

Change-Id: I0d1b289858d885e3eeff8ec8e08a45eee5419357
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 9ccd306..293d916 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -52,6 +52,7 @@
 }
 
 #define MEM_CPY_ERRATA_WR_OS_VAL		62	/* wr os should be 62 as per HW errata GAUDI3_0315 */
+#define MEM_CPY_ERRATA_RD_OS_VAL		0	/* rd os should be 0 as per HW errata GAUDI3_0306 */
 int arcboot_rom_mem_init(uint64_t addr, uint32_t size)
 {
 	uint64_t start = arcboot_rom_get_ts_us();
@@ -73,6 +74,11 @@
 	 */
 	mem_cfg |= (MEM_CPY_ERRATA_WR_OS_VAL << D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_WR_OS_S);
 
+	/*
+	 * WA for HW errata GAUDI3_0306
+	 */
+	mem_cfg |= (MEM_CPY_ERRATA_RD_OS_VAL << D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_RD_OS_S);
+
 	regmap_writel(mem_cfg, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
 	regmap_writel(0x0, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CONST);
 
